Add USB HID tests

fkjagodzinski opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
Add `tests-usb_device-hid` test suite for `USBHID`, `USBMouse` & `USBKeyboard`.

### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->
@c1728p9, @jamesbeyond, @maciejbocianski 

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

fkjagodzinski committed:
USBHID: Fix the initial HID report read operation

The first 4 bytes received were lost due to a wrong address.
Read the output report into HID_REPORT.data.

fkjagodzinski committed:
Tests: USBHID: Add tests

fkjagodzinski commented on requirements.txt:
```diff
@@ -23,4 +23,5 @@ git+https://github.com/armmbed/manifest-tool.git@v1.4.6
 icetea>=1.2.1,<1.3
 pycryptodome>=3.7.2,<=3.7.3
 pyusb>=1.0.0,<2.0.0
+hidapi>=0.7.99,<0.8.0
```
@c1728p9, @jamesbeyond I'd like to clarify the licensing of this module. Based on its [LICENSE file](https://github.com/trezor/cython-hidapi/blob/master/LICENSE.txt) it can be used under one of 3 licenses. Should I specify, which one we want to use? If so, how do I do that?

cmonr replied:
@0xc0170 Might be a question for you as well

0xc0170 replied:
I need more context here, one of 3 licenses? Does this module hidapi have dual license or?

fkjagodzinski replied:
Yes 3
> cython-hidapi may be used by one of three licenses as outlined in LICENSE.txt

0xc0170 replied:
Got it . This is missing in our license file (capturing python dependencies there!) , can you fix that ? IT would be added there as well - hidapi under BSD license

0xc0170 replied:
Created #10123

fkjagodzinski committed:
Tests: USBHID: Reuse VID & PID from basic test

To successfully use pyusb on Windows hosts, a Zadig configuration has to
be performed. Since config for basic tests has already been provided,
use it again.

c1728p9 commented on TESTS/usb_device/hid/main.cpp:
```diff
@@ -0,0 +1,376 @@
+    usb_hid.connect();
+    greentea_send_kv(MSG_KEY_TEST_RAW_IO, REPORT_SIZE);
+    usb_hid.wait_ready();
+    wait_ms(500);
```
Why the delay?

fkjagodzinski replied:
I added that temporarily to handle host side driver setup. It turned out to take much longer on Windows than Linux. It is replaced with a `greentea_parse_kv()` now.

fkjagodzinski committed:
Tests: USBHID: Handle variable driver setup time

Wait for the host driver to finish setup before sending any HID reports
from the device.

USBHID::wait_ready() blocks until the device reaches 'configured' state,
but the state of the host HID driver remains unknown to the device.

fkjagodzinski committed:
Tests: USBHID: Remove unnecessary wait_ready call

Every test case waits at greentea_parse_kv() anyway.

fkjagodzinski committed:
Tests: USBHID: Make report test optional on Linux

This test case uses `hidapi` -- a cross-platform Python module.
To keep the initial Mbed setup as simple as possible, the `hidapi`
module is skipped on Linux hosts because of its external dependancies
for this platform.

The module can be easily installed following instructions from the
README file.
The test case is skipped if the host machine lacks `hidapi` module.

fkjagodzinski committed:
Tests: USBHID: Specify `hidapi` license
